src/base/bittorrent/peerinfo.cpp

```diff
@@ -230,24 +230,14 @@ void PeerInfo::calcRelevance(const Torrent *torrent)
 {
     const QBitArray allPieces = torrent->pieces();
     const QBitArray peerPieces = pieces();
+    Q_ASSERT(allPieces.size() == peerPieces.size());
 
-    int localMissing = 0;
-    int remoteHaves = 0;
+    const int remoteHaves = ((~allPieces) & peerPieces).count(true);
+    const int localMissing = allPieces.count(false);
```

Comment:
I would calculate it first and return if it is zero.

Reply:
Done.